app/utils.js

```diff
@@ -129,6 +129,73 @@ define(['underscore'], function (_) {
     return value == null || value === '';
   };
 
+  /**
+   * Returns a function, that, as long as it continues to be invoked, will not be triggerd.
+   * The function will be called after it stops being called for N ms.
+   * Based on https://github.com/component/debounce/blob/master/index.js which in turn is based on the
+   * original implemenation in Underscore (both licensed MIT)
+   */
+  Utils.debounce = function(func, wait, immediate) {
```

Comment:
Are we still going to use this debounce? We could use the underscore implementation